app/src/main/java/tech/ula/ui/AppListFragment.kt

```diff
@@ -86,6 +96,18 @@ class AppListFragment : Fragment() {
                 is AppItem -> {
                     val selectedApp = selectedItem.app
                     doAppItemClicked(selectedApp)
+
+                    val preferredServiceType = appListViewModel.getAppServiceTypePreference(appList[position])
```

Comment:
appList[position] is broken now. There are more items than apps in the list because of the separators. Use `selectedApp`.